soh/src/code/z_message_PAL.c

```diff
@@ -3417,7 +4577,11 @@ void Message_Update(PlayState* play) {
                     if (msgCtx->textboxEndType == TEXTBOX_ENDTYPE_HAS_NEXT) {
                         Audio_PlaySoundGeneral(NA_SE_SY_MESSAGE_PASS, &gSfxDefaultPos, 4, &gSfxDefaultFreqAndVolScale, &gSfxDefaultFreqAndVolScale,
                                                &gSfxDefaultReverb);
-                        Message_ContinueTextbox(play, sNextTextId);
+                        if (gSaveContext.language == LANGUAGE_JPN && !sTextIsCredits && !sDisplayNextMessageAsEnglish) {
+                            Message_ContinueTextbox(play, msgCtx->msgBufDecodedWide[msgCtx->textDrawPos]);
+                        } else {
+                            Message_ContinueTextbox(play, sNextTextId);
+                        }
```

Comment:
With japanesse set, this line always causes a crash because the textID is not a real ID. I tried using `sNextTextId` instead and the crashing stopped. I can't read japanesse, but it at least "seems" correct. Not sure why you had to have it like this in the first place considering your `Jpn` funcs do set `sNextTextId`.
```suggestion
                        Message_ContinueTextbox(play, sNextTextId);
```

Reply:
Hmm, I see your code matches what decomp has. Perhaps the msgBufDecoded has incorrect data, or something else. Not sure what should be done.

Reply:
Ok it looks like the problem is due to Fast Text, it appears to be incrementing the `msgCtx->textDrawPos` to far, causing the wrong buf value to be read.

Reply:
Hm, how do you have your settings set up? I can't seem to get any kind of crash

Reply:
I just had Fast Text set to 5 and talked to a kokiri that had a continue text box (like fresh debug save). The one outside Saria's house or ontop of the shop that auto talks to you for example.